models/user-activate.php

```diff
@@ -23,70 +23,114 @@ class UserActivate extends \DustPress\Model {
-                            __( 'Your account has been activated. You may now <a href="%1$s">log in</a> to the site using your chosen username of &#8220;%2$s&#8221;. Please check your email inbox at %3$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%4$s">reset your password</a>.' ), network_site_url( 'wp-login.php', 'login' ), $signup->user_login, $signup->user_email, wp_lostpassword_url() );
-                    }
-                    else {
-                        // site active and email sent
-                        $state = "site-active-mail";
-                        /* translators: 1: site URL, 2: site domain, 3: username, 4: user email, 5: lost password URL */
-                        $this->print['message'] = sprintf( /* translators: 1: site URL, 2: site domain, 3: username, 4: user email, 5: lost password URL */
-                            __( 'Your site at <a href="%1$s">%2$s</a> is active. You may now log in to your site using your chosen username of &#8220;%3$s&#8221;. Please check your email inbox at %4$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%5$s">reset your password</a>.' ), 'http://' . $signup->domain, $signup->domain, $signup->user_login, $signup->user_email, wp_lostpassword_url() );
-                    }
+
+            if ( is_wp_error( $result ) && in_array( $result->get_error_code(), $valid_error_codes ) ) {
+                $signup = $result->get_error_data();
+
+                $this->signup = $signup;
+                $this->print['title'] = apply_filters( 'dustpress/activate/account_active',  __( 'Your account is now active!' ) );
+
+                if ( $signup->domain . $signup->path == '' ) {
+                    // account active and email sent
+                    $state = "account-active-mail";
+                    $this->print['message'] = apply_filters(
+                        'dustpress/activate/user_activated',
+                        sprintf( /* translators: 1: login URL, 2: username, 3: user email, 4: lost password URL */
+                            __( 'Your account has been activated. You may now <a href="%1$s">log in</a> to the site using your chosen username of &#8220;%2$s&#8221;. Please check your email inbox at %3$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%4$s">reset your password</a>.' ),
+                            network_site_url( 'wp-login.php', 'login' ),
+                            $signup->user_login,
+                            $signup->user_email,
+                            wp_lostpassword_url()
```

Comment:
wp_lost_password_url can be filtered, should it be escaped with esc_url?

Reply:
These are all for the messages that are shown to the user and not for any functionality, so I'd keep them unescaped. If the developer wants to output something not escaped, it should be possible.

Reply:
But the code above outputs a link to reset a user's password. Am I wrong in thinking a compromised plugin could filter that URL to include e.g. some JS? I don't see how this is different from line 115: `esc_url( $login_url )` except that there's a direct function call instead of a variable.

Reply:
This is supposed to mimic the WordPress core as strictly as possible, and the core uses the same value unescaped. You have a point, but it could also break some functionality for another plugin compared to what the core does.

Reply:
HMMM.

Reply:
https://github.com/WordPress/WordPress/blob/master/wp-activate.php#L152